src/components/CreateRecipeModal.tsx

```diff
@@ -59,50 +60,63 @@ const CreateRecipeModal: FC<Props> = ({ isOpen, onClose, focusRef }) => {
   if (isError) {
     return <div>Error</div>;
   }
+
   return (
-    <Modal initialFocusRef={focusRef} isOpen={isOpen} onClose={onClose}>
-      <ModalOverlay />
-      <ModalContent>
-        <ModalHeader>
-          {showInput ? "Vyberte vhodné produkty" : "Zadejte název ingredience"}
-        </ModalHeader>
-        <ModalCloseButton />
-        <ModalBody pb={6}>
-          {showInput ? (
-            <>
-              <CreateRecipeInput
-                searchQuery={searchQuery}
-                setSearchQuery={setSearchQuery}
-                productsByIds={productsByIds}
-                productIds={productIds}
-                selectedProducts={selectedProducts}
-                setSelectedProducts={setSelectedProducts}
-              />
-              {Object.keys(selectedProducts).map((searchQuery) => {
-                const products = selectedProducts[searchQuery];
+    <>
+      <div ref={modalContainer}></div>
+      <Modal
+        portalProps={{
+          containerRef: modalContainer,
```

Comment:
Zjistil jsem, ze under the hood to pouziva jejich komponentu Portal. https://chakra-ui.com/docs/components/modal/props pak jsem nasel tuhle propu, takze tomu jen vstuneme nas "container" kterej zije v nasem ShadowDomu, tohle by melo zatim stacit.